src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -1050,6 +1050,7 @@ private void LogFailedProbe(SiloAddress silo, int pingNumber, Exception failureR
 
                 if (logger.IsVerbose) logger.Verbose("-Going to DeclareDead silo {0} in the table. About to write entry {1}.", entry.SiloAddress.ToLongString(), entry.ToFullString());
                 entry.Status = SiloStatus.Dead;
+                RuntimeClient.Current.BreakOutstandingMessagesToDeadSilo(entry.SiloAddress);
```

Comment:
That it is a different, unrelated bug in `GrainBasedMembershipTable`. I will look into it later on. local subscribers should be notified.